keps/sig-apps/2804-consolidate-workload-controllers-status/README.md

```diff
@@ -0,0 +1,808 @@
+-->
+- Upgrades
+ When upgrading from a release without this feature, to a release with 
+ `progressDeadlineSeconds`, we will set `progressDeadlineSeconds` to max value of int32 (i.e. 2147483647). This would give users 
+ the same default behavior.
+- Downgrades
+ When downgrading from a release with this feature, to a release without 
+ `progressDeadlineSeconds`, there are two cases
+  - If `progressDeadlineSeconds` is greater than 0 -- in this case kube-apiserver
+    clears the `progressDeadlineSeconds` and the existing StatefulSet and DaemonSet controllers wouldn't honor 
+    `progressDeadlineSeconds` which is expected.
+  - If `progressDeadlineSeconds` is equal to 0 -- in this case user wont see any 
+    difference in behavior.
+
+We will ensure that the `progressDeadlineSeconds` field is properly validated
+before persisting. The validation includes checking for positive number for the `progressDeadlineSeconds` field.
+
+### Version Skew Strategy
+
+<!--
+If applicable, how will the component handle version skew with other
+components? What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- Does this enhancement involve coordinating behavior in the control plane and
+  in the kubelet? How does an n-2 kubelet without this feature available behave
+  when this feature is used?
+- Will any other components on the node change? For example, changes to CSI,
+  CRI or CNI may require updating that component before the kubelet.
+-->
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+-->
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: DaemonSetConditions, StatefulSetConditions
```

Comment:
Are there two new feature flags? Why?

Reply:
There should be just one which will apply to all controllers.

Reply:
I think the idea behind this is to have better granularity for `progressDeadlineSeconds` in DaemonSet and StatefulSet since this change could be unexpected to some users and they might want to opt in only for one of these.